File: solutions/0121-best-time-to-buy-and-sell-stock/solution.js
/**
 * @param {number[]} price
 * @return {number}
 */
var maxProfit = function(price) {
  let min = price[0];
  let profit = 0;
  for(let i = 0;i<price.length;i++){
    if(price[i] < min){
        min = price[i];
    }
    else{
        profit =  Math.max(profit,price[i] - min)
    }
  }
    return profit;
}

File: solutions/0121-best-time-to-buy-and-sell-stock/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minprice = INT_MAX;
        int maxprofit = 0;
        for(int price : prices){
            minprice = min(minprice,price);
            int profit = price - minprice;
            maxprofit = max(maxprofit,profit);
        }
        return maxprofit;
    }
};
